src/Browser/Avalonia.Browser/Interop/CanvasHelper.cs

```diff
@@ -1,25 +1,23 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Runtime.CompilerServices;
-using System.Runtime.InteropServices;
-using System.Runtime.InteropServices.JavaScript;
+﻿using System.Runtime.InteropServices.JavaScript;
+using System.Threading.Tasks;
+using Avalonia.Threading;
 
 namespace Avalonia.Browser.Interop;
 
 internal record GLInfo(int ContextId, uint FboId, int Stencils, int Samples, int Depth);
 
 internal static partial class CanvasHelper
 {
-    [JSImport("CanvasSurface.onSizeChanged", AvaloniaModule.MainModuleName)]
-    public static partial void OnSizeChanged(
-        JSObject canvasSurface,
-        [JSMarshalAs<JSType.Function<JSType.Number, JSType.Number, JSType.Number>>]
-        // TODO: this callback should be <int, int, double>. Revert after next .NET 9 preview.  
-        Action<double, double, double> onSizeChanged);
+    [JSExport]
+    public static Task OnSizeChanged(int topLevelId, double width, double height, double dpr)
```

Comment:
I think that we are supposed to dispatch this synchronously in ST mode, maybe via Dispatcher.Invoke

Reply:
+